geode-benchmarks/src/main/java/org/apache/geode/benchmark/parameters/GcParameters.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.benchmark.parameters;
+
+import static org.apache.geode.benchmark.parameters.JavaVersion.v11;
+import static org.apache.geode.benchmark.parameters.Utils.configureAll;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.geode.perftest.TestConfig;
+
+public class GcParameters {
+  private static final Logger logger = LoggerFactory.getLogger(GcParameters.class);
+
+  public static void configure(final TestConfig testConfig) {
+    final JavaVersion javaVersion = JavaVersion.current();
+    final GcImplementation gcImplementation =
+        GcImplementation.valueOf(System.getProperty("withGc", "CMS"));
+    logger.info("Configuring {} GC.", gcImplementation);
+    switch (gcImplementation) {
+      case CMS:
+        configureAll(testConfig,
+            "-XX:+UseConcMarkSweepGC",
+            "-XX:+UseCMSInitiatingOccupancyOnly",
+            "-XX:+CMSClassUnloadingEnabled",
+            "-XX:+CMSScavengeBeforeRemark",
+            "-XX:CMSInitiatingOccupancyFraction=60",
+            "-XX:+UseNUMA",
+            "-XX:+ScavengeBeforeFullGC",
+            "-XX:+UnlockDiagnosticVMOptions",
+            "-XX:ParGCCardsPerStrideChunk=32768");
+        break;
+      case G1:
+        configureAll(testConfig,
+            "-XX:+UseG1GC",
+            "-XX:+UseNUMA",
+            "-XX:+ScavengeBeforeFullGC");
+        break;
+      case Z:
+        if (javaVersion.olderThan(v11)) {
+          throw new IllegalArgumentException("ZGC requires Java 11 or newer");
+        }
+        configureAll(testConfig,
+            "-XX:+UnlockExperimentalVMOptions",
+            "-XX:+UseZGC");
+        // if (javaVersion.atLeast(v13)) {
```

Comment:
Oh that's a good catch. I put this in before I made the heap configurable. So having a fixed value here made since up until that point. I wonder if now we should have an occupancy threshold configuration option, similar to that of CMS, that calculates the soft max for Z and just sets the value for CMS. @metatype do you have any thoughts on this?